server/src/routes/api/v1/userSaveFileRouter.js

```diff
@@ -17,23 +22,75 @@ userSaveFileRouter.get("/:id", async (req, res) => {
 
 userSaveFileRouter.post("/:id", async (req, res) => {
   const body = req.body
-  const puzzlesId = req.params.id
+  const puzzleId = req.params.id
   const userId = req.user.id
+  let correctlySolved = false
   
-  const formInput = cleanUserInput(body)
+  const formInput = cleanUserSubmission(body)
   let serializedformInput = JSON.stringify(formInput);
 
   let userSave = {}
   userSave.savedPuzzle = serializedformInput
   userSave.userId = userId
-  userSave.puzzlesId = puzzlesId
+  userSave.puzzleId = puzzleId
 
-  try {
-    const newSave = await UserSaveFile.query().insertAndFetch(userSave)
-    return res.status(200).json({ userSaveFile: newSave })
-  } catch(error) {
-    return res.status(500).json({ error })
+  const user = await User.query().findById(userId)
+  const userSaveFile = await user.$relatedQuery("userSaveFile")
+
+  let matchedSave = false
+  userSaveFile.forEach(save => {
+    if(save.puzzleId === puzzleId){
+      matchedSave = save.id
+    }
+  })
+
+  if(matchedSave === false){
+    try {
+      const newSave = await UserSaveFile.query().insertAndFetch(userSave)
+      const checkPuzzle = new checkSubmission(formInput)
+
+      if(checkPuzzle.isCorrect){
+          correctlySolved = true
+        }
+        else{
+          correctlySolved = false
+        }
+        console.log(checkPuzzle.isCorrect)
+
+      return res.status(200).json({ userSaveFile: correctlySolved })
+    } catch(error) {
+      if(error instanceof ValidationError) {
+        return res.status(422).json({errors: error.data})
+      }
+      return res.status(500).json({ error })
+    }
+  }
+  else{
+    try {
+      const updateSave = await UserSaveFile.query().updateAndFetchById(matchedSave, {
+        savedPuzzle: userSave.savedPuzzle
+      })
+      const checkPuzzle = new checkSubmission(formInput)
+      console.log(checkPuzzle)
+
+      if(checkPuzzle.isCorrect){
+          correctlySolved = true
+        }
+        else{
+          correctlySolved = false
+        }
+        console.log(checkPuzzle.isCorrect)
+
+      return res.status(200).json({ userSaveFile: correctlySolved })
+    } catch(error) {
+      if(error instanceof ValidationError) {
+        return res.status(422).json({errors: error.data})
+      }
+      return res.status(500).json({ error })
+    }
```

Comment:
you dont need two try/catch blocks in this endpoint. You can have one, and then have the if/else based on the value of `matchedSave` inside of the try/catch. The catch can catch validation errors from either the insert or the update